### PR TITLE
Fix duplicate symbols handling in SetNode symbols unaliasing

### PR DESCRIPTION
**Pull request by sopel39**
quick fix for: https://github.com/prestodb/presto/issues/6778

### CODE REVIEW

**Comment by martint** [comment]
Can you add a test?

**Comment by sopel39** [comment]
Sure, morning CET I will minimal query that fails

**Comment by sopel39** [comment]
Actually I have added test now. Back to you

File: presto-main/src/main/java/com/facebook/presto/sql/planner/optimizations/UnaliasSymbolReferences.java [hunk 1]
```diff
@@ -695,10 +695,12 @@ public Expression rewriteSymbolReference(SymbolReference node, Void context, Exp
                     orderings.build());
         }
 
-        private List<Symbol> canonicalize(List<Symbol> symbols)
+        private List<Symbol> canonicalizeUnique(List<Symbol> symbols)
```
Comment (raghavsethi) [inline presto-main/src/main/java/com/facebook/presto/sql/planner/optimizations/UnaliasSymbolReferences.java]: How is this different from canonicalizeAndDistinct?

**Comment by sopel39** [comment]
applied comments & added more tests

**Comment by raghavsethi** [comment]
Merged, thanks!

**Pull request closed without merging**